### PR TITLE
Scripting: JSON parsing and writing in watcher

### PR DESCRIPTION
**Pull request by stu-elastic**
* `Json.load(String)`
* `Json.dump(Object)`
* `Json.dump(Object, boolean)`.

### CODE REVIEW

**Review by jdconrad — APPROVED** [review]
LGTM. Would prefer @rjernst look too since I wrote a few of the tests.

**Comment by jdconrad** [comment]
@stu-elastic Thanks for separating the docs out into a different PR. Still LGTM.

**Comment by stu-elastic** [comment]
@elasticmachine run elasticsearch-ci/2

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Json.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+
+import java.io.IOException;
+
+public class Json {
+    public static Object load(String json) throws IOException{
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Json.java:31]: javadocs?

**Review by rjernst — APPROVED** [review]
This looks good, but we also need to consider how this is documented for users. There should be something in the painless reference or watcher docs?

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Json.java (hunk 1 above)
Comment (stu-elastic) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Json.java:31]: Painless reference is currently broken, which is why I haven't updated context docs. We will address that separately.

Comment (stu-elastic) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/api/Json.java:31]: Added javadocs.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by stu-elastic** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by stu-elastic** [comment]
Master (8.0): f43e52d
7.x (7.10): 8a61b95

**Comment by agone1** [comment]
Hello @stu-elastic !

Is there any chance that same functionality would be added to other painless context beside watcher?

**Comment by djmcgreal-cc** [comment]
Seems weird just to have it in Watcher. Specifically I was hoping to use it in painless execute API to allow unit testing of the result. Current behaviour prints out a Java .toString which is pretty impossible to parse correctly in clients.

**Comment by stu-elastic** [comment]
Hi @agone1 and @djmcgreal-cc.

We're being cautious about rolling out this feature in other contexts because parsing JSON can create a lot of objects leading to GC pressure if it's run in, say, an aggregation.

@agone1 What contexts are you interested in?  Can you cut a new issue for discussion?

@djmcgreal-cc Good point, I've added it (and some other per-context whitelists) to the execute api in https://github.com/elastic/elasticsearch/issues/67035.

**Pull request merged**